metadata-ingestion/src/datahub/ingestion/source/redash.py

```diff
@@ -639,11 +656,12 @@ def _emit_chart_mces(self) -> Iterable[MetadataWorkUnit]:
             for query_response in queries_response["results"]:
 
                 chart_name = query_response["name"]
-
+                database_name = self._get_database_name_from_datasource_id(query_response["data_source_id"])
```

Comment:
question - is query_response guaranteed to have the "data_source_id" field?

Reply:
Every query_response has a data_source_id even if its a empty query, so Yes!

Reply:
Amazing - thanks :)